lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1320,17 +1418,22 @@ def _update_relation_data(self, _):
         if not self._charm.unit.is_leader():
             return
 
-        if alert_groups := load_alert_rules_from_dir(
-            self.dir_path, self.topology, recursive=self._recursive
-        ):
+        rules_file = (
+            AlertRules(self.topology)
+            .add_from_path(self.dir_path, recursive=self._recursive)
+            .as_dict()
```

Comment:
1. It is not very intuitive to see and `AlertRules` object being instantiated with a `topology` argument and then the rules themselves being added using a method. It is not clear why the `AlertRules` can not be instantiated with both a alert rules path and the topology.
2. It is not very intuitive to see an `AlertRules` converted to a `dict` and assigned to a variable called `rules_file` instead of just something like `alert_rules`. A `file` suffix is suggests that this is a file object.

Reply:
Refactored.